flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -726,6 +730,41 @@ void sendPartitionInfos() {
 		}
 	}
 
+	/**
+	 * Check whether the InputDependencyConstraint is satisfied for this vertex.
+	 *
+	 * @return whether the input constraint is satisfied
+	 */
+	public boolean checkInputDependencyConstraints() {
+		if (getExecutionGraph().getInputDependencyConstraint() == InputDependencyConstraint.ANY) {
```

Comment:
Shouldn't the `InputDependencyConstraint` rather be a value of the `ExecutionJobVertex` than the `ExecutionGraph`? I guess it should be configurable for each operator individually.

Reply:
I've move `InputDependencyConstraint ` to JobVertex. And the job wide default value can be configured in `ExecutionConfig`. But I haven't make it configurable through DataSet/DataStream API yet.

I agree we should support the constraint configurable for each operator. But I'm not quite sure whether we should support it with DataSet API or later for the unified StreamGraph/Transformation API? Could you share your suggestion?

In our production experience, a job-wide configured input constraint satisfies most users, together with `BATCH_FORCED` execution mode, to ensure a batch job can finish with limited resources.